File: app/src/main/java/rcdavis/nycschools/util/RetrofitUtils.java
package rcdavis.nycschools.util;

import hu.akarnokd.rxjava3.retrofit.RxJava3CallAdapterFactory;
import okhttp3.OkHttpClient;
import rcdavis.nycschools.BuildConfig;
import rcdavis.nycschools.school.SchoolAPI;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitUtils {
    public static SchoolAPI createSchoolApi() {
        return createApi(SchoolAPI.class, BuildConfig.SCHOOL_API_URL);
    }

    public static <T> T createApi(final Class<T> tClass, final String baseUrl) {
        return new Retrofit.Builder()
                .baseUrl(baseUrl)
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
                .client(new OkHttpClient.Builder().build())
                .build()
                .create(tClass);
    }
}

File: app/src/main/java/rcdavis/nycschools/util/RetrofitUtils.kt
package rcdavis.nycschools.util

import hu.akarnokd.rxjava3.retrofit.RxJava3CallAdapterFactory
import okhttp3.OkHttpClient
import rcdavis.nycschools.BuildConfig
import rcdavis.nycschools.school.SchoolAPI
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitUtils {
    @JvmStatic fun createSchoolApi(): SchoolAPI {
        return createApi(SchoolAPI::class.java, BuildConfig.SCHOOL_API_URL)
    }

    @JvmStatic fun <T> createApi(tClass: Class<T>, baseUrl: String): T {
        return Retrofit.Builder()
            .baseUrl(baseUrl)
            .addConverterFactory(GsonConverterFactory.create())
            .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
            .client(OkHttpClient.Builder().build())
            .build()
            .create(tClass)
    }
}
